shuttle_test.go

```diff
@@ -98,7 +101,60 @@ func TestIntegration(t *testing.T) {
 	if afterDrops, _ := shut.Drops.ReadAndReset(); afterDrops != 0 {
 		t.Fatalf("afterDrops=%d\n", afterDrops)
 	}
+}
+
+func TestSocketListener(t *testing.T) {
+	th := new(testHelper)
+	ts := httptest.NewServer(th)
+	defer ts.Close()
+
+	config.LogsURL = ts.URL
+	config.Socket = "/tmp/log-shuttle.sock"
+
+	// Assert that the socket does not exist
+	if _, err := os.Stat(config.Socket); err == nil {
+		t.Fatalf("exists=%s\n", config.Socket)
+	}
+
+	shut := NewShuttle(config)
+	shut.Launch()
+
+	// Give the socket time to be created.
+	time.Sleep(300 * time.Millisecond)
+
+	// Confirm the socket is created
+	if _, err := os.Stat(config.Socket); err != nil {
+		t.Fatalf("missing=%s\n", config.Socket)
+	}
+
+	// Connect to the socket
+	c, err := net.Dial("unix", config.Socket)
+	if err != nil {
+		t.Fatalf("connect_failed=%s\n", config.Socket)
+	}
+
+	// Write to the socket
+	c.Write([]byte("a log line\n"))
+	c.Write([]byte("another log line\n"))
+
+	// Give the socket time to write to the outlet
+	time.Sleep(1 * time.Second)
+
+	shut.Land()
+
+	pat1 := regexp.MustCompile(`a log line`)
```

Comment:
This is handled in the README example by forcing input-format=1 (PS: I probably need to add a test for this as I'm not 100% sure it actually still works), but that applies to all data passing through, no matter the source (syslog or stdin).

We would have to pass that expected input format to the reader, which would then need to tag each LogLine with the starting format, so that the formatters down the line can do the right thing for output to logplex (or other destinations).